react-components/src/library/components/signup/teacher_form.js

```diff
@@ -68,6 +68,13 @@ export default class TeacherForm extends React.Component {
   }
 
   submit (data, resetForm, invalidateForm) {
+    // the updated react-select element uses {label, value} as the value so pull the value (id) from it
+    if (data.country_id && data.country_id.value) {
+      data.country_id = data.country_id.value
+    }
+    if (data.school_id && data.school_id.value) {
+      data.school_id = data.school_id.value
+    }
```

Comment:
Is there any danger in modifying the input `data` object? Perhaps the code calling submit is counting on the data object not being changed by the `submit` function?

Reply:
Looking at the code as is there doens't look to be a problem but I'll move it after the jQuery.extend() call so that I'm operating on a fresh object.

Reply:
@scytacki I've added a commit to resolve the issue you found.